components/dialog/dialog-mixin.js

```diff
@@ -453,6 +454,8 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 			else await animPromise;
 
 			/** Dispatched when the dialog is opened */
+			await this.#waitForUpdateComplete();
+			await this._updateSize();
```

Comment:
I still think we should wrap this in a global client-side feature flag.

Reply:
We'll likely also want to give some kind of heads-up on `#web-platform` as I'd be surprised if this extra waiting doesn't affect some vdiffs -- likely in a "good" way, but potentially could still break stuff.